scripts/mac-precheck.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# A script to check if the necessary requirements for running the KinD partner cluster on Mac actually exist.
+
+# Check if Docker is installed
+CHECK_PODMAN=false
+if ! [ -x "$(command -v docker)" ]; then
+
+	# Check if podman is installed in a subsequent step
+	if ! [ -x "$(command -v podman)" ]; then
+		CHECK_PODMAN=true
+	fi
+
+  # Default to recommend docker if podman is not installed
+	if [ "$CHECK_PODMAN" == "false" ]; then
+		echo "Docker is not installed. Please install Docker Desktop for Mac and try again."
+		exit 1
+	fi
+else 
+  echo "Docker is installed. $(docker --version)" 
+fi
+
+# Check if podman is installed and KIND_EXPERIMENTAL_PROVIDER is set in the environment
+if [ "$CHECK_PODMAN" == "true" ]; then
+	if ! [ -x "$(command -v podman)" ]; then
```

Comment:
You already checked podman, this condition never happens.

Reply:
Fixed.